frontend/lib/resolution/Resolver.cpp

```diff
@@ -1949,7 +1951,42 @@ void Resolver::handleResolvedCallPrintCandidates(ResolvedExpression& r,
       if (!rejected.empty()) {
         // There were candidates but we threw them out. We can issue a nicer
         // error explaining why each candidate was rejected.
-        CHPL_REPORT(context, NoMatchingCandidates, call, ci, rejected);
+        std::vector<resolution::FormalActual> badPasses;
+        std::vector<const uast::VarLikeDecl*> actualDecls;
+        badPasses.resize(rejected.size());
+        actualDecls.resize(rejected.size());
+        // check each rejected candidate for uninitialized actuals
+        for (size_t i = 0; i < rejected.size(); i++) {
+          auto& candidate = rejected[i];
+          if (candidate.reason() == resolution::FAIL_CANNOT_PASS &&
+              /* skip printing detailed info_ here because computing the formal-actual
+              map will go poorly with an unknown formal. */
+              candidate.formalReason() != resolution::FAIL_UNKNOWN_FORMAL_TYPE) {
+            auto fn = candidate.initialForErr();
+            resolution::FormalActualMap fa(fn, ci);
+            auto badPass = fa.byFormalIdx(candidate.formalIdx());
```

Comment:
Consider & here if possible to avoid double copy (into badPss then into badPasses[i])